rosetta/Dockerfile.pax

```diff
@@ -2,55 +2,73 @@
+# the way up to the jax build.
+ARG UPDATE_PATCHES=false
 
-FROM scratch as rosetta-source
-ARG SRC_PATH_PAXML
-ARG SRC_PATH_PRAXIS
-
-COPY . /
-
-FROM scratch as pax-mirror-source
-ADD --keep-git-dir=true https://github.com/google/paxml.git#main /
-
-FROM scratch as praxis-mirror-source
-ADD --keep-git-dir=true https://github.com/google/praxis.git#main /
+# Rosetta and optionally patches are pulled from this
+FROM scratch AS jax-toolbox
 
 ###############################################################################
 ### Download source and add auxiliary scripts
 ################################################################################
 
 FROM ${BASE_IMAGE} AS mealkit
-ENV ENABLE_TE=1
-
 ARG GIT_USER_EMAIL
 ARG GIT_USER_NAME
-ARG PAXML_PATCHLIST
-ARG PRAXIS_PATCHLIST
+ARG UPDATE_PATCHES
 
-COPY --from=rosetta-source / /opt/rosetta
-WORKDIR /opt/rosetta
-RUN --mount=target=/opt/pax-mirror,from=pax-mirror-source,readwrite \
-    --mount=target=/opt/praxis-mirror,from=praxis-mirror-source,readwrite <<EOF bash -e
+ENV ENABLE_TE=1
+
+RUN --mount=target=/mnt/jax-toolbox,from=jax-toolbox <<"EOF" bash -exu
+############DELETE
 git config --global user.email "${GIT_USER_EMAIL}"
 git config --global user.name "${GIT_USER_NAME}"
 
-bash create-distribution.sh \
-  -p ${PAXML_PATCHLIST} \
-  -m https://github.com/nvjax-svc-0/paxml.git \
-  -d /opt/paxml \
-  -e /opt/pax-mirror
-bash create-distribution.sh \
-  -p ${PRAXIS_PATCHLIST} \
-  -m https://github.com/nvjax-svc-0/praxis.git \
-  -d /opt/praxis \
-  -e /opt/praxis-mirror
+if [[ ! -d /opt/pip ]]; then
+  git clone https://github.com/pypa/pip.git /opt/pip
+fi
+git -C /opt/pip checkout 23.3.1
+cp /mnt/jax-toolbox/.github/container/pip-vcs-equivalency.patch /opt/pip/
+cd /opt/pip
+git apply </opt/pip/pip-vcs-equivalency.patch
+git add -u
+git commit -m 'Adds JAX_TOOLBOX_VCS_EQUIVALENCY as a trigger to treat all github VCS installs for a package as equivalent. The spec of the last encountered version will be used'
+pip install --upgrade --no-cache-dir /opt/pip pip-tools && rm -rf ~/.cache/*
+cd -
+############DELETE
+
+MANIFEST_DIR=$(dirname ${MANIFEST_FILE})
+if [[ "${UPDATE_PATCHES}" != "true" && "${UPDATE_PATCHES}" != "false" ]]; then
+  echo "UPDATE_PATCHES can only be true or false"
+  exit 1
+fi
+if [[ "${UPDATE_PATCHES}" == "true" ]]; then
+  cp -r /mnt/jax-toolbox/.github/container/patches ${MANIFEST_DIR}/patches
+  cp /mnt/jax-toolbox/.github/container/manifest.yaml ${MANIFEST_DIR}/manifest.yaml
+  cp /mnt/jax-toolbox/.github/container/create-distribution.sh ${MANIFEST_DIR}/create-distribution.sh
+  # TODO: remove
+  cp /mnt/jax-toolbox/.github/container/pip-finalize.sh /usr/local/bin/
+fi
+cp -r /mnt/jax-toolbox/rosetta /opt/rosetta
+
+# Setting the username/email is required to author commits from patches
+git config --global user.email "${GIT_USER_EMAIL}"
+git config --global user.name "${GIT_USER_NAME}"
+
+bash ${MANIFEST_DIR}/create-distribution.sh \
```

Comment:
It fails, if UPDATE_PATHCES == false, with error `create-disctribution.sh is not found`

Reply:
Could you share how you ran this? I can look into this